py/picca/delta_extraction/astronomical_objects/forest.py

```diff
@@ -122,73 +121,78 @@ def rebin(log_lambda, flux, ivar, transmission_correction, z, wave_solution,
                10**log_lambda_rest_frame_grid[0] - half_pixel_step_rest_frame)
         w1 &= (lambda_ / (1. + z) <
                10**log_lambda_rest_frame_grid[-1] + half_pixel_step_rest_frame)
-        w1 &= (ivar > 0.)
+        # w1 &= (ivar > 0.)
     else:
         raise AstronomicalObjectError("Error in Forest.rebin(). "
                                       "Class variable 'wave_solution' "
                                       "must be either 'lin' or 'log'.")
 
+    if (w1 & (ivar > 0.)).sum() == 0:
+        log_lambda = np.zeros(0)
+        flux = np.zeros(0)
+        ivar = np.zeros(0)
+        transmission_correction = np.zeros(0)
+        mean_snr = 0.0
+        bins = np.zeros(0, dtype=np.int64)
+        rebin_ivar = np.zeros(0)
+        w1 = np.zeros(0, dtype=bool_)
+        w2 = np.zeros(0, dtype=bool_)
+        return (log_lambda, flux, ivar, transmission_correction, mean_snr, bins,
+                rebin_ivar, orig_ivar, w1, w2)
+
     log_lambda = log_lambda[w1]
     flux = flux[w1]
     ivar = ivar[w1]
     transmission_correction = transmission_correction[w1]
-    if w1.sum() == 0:
-        log_lambda = np.zeros(log_lambda.size)
-        flux = np.zeros(log_lambda.size)
-        ivar = np.zeros(log_lambda.size)
-        transmission_correction = np.zeros(log_lambda.size)
-        mean_snr = 0.0
-        bins = np.zeros(log_lambda.size, dtype=np.int64)
-        rebin_ivar = np.zeros(log_lambda.size)
-        w1 = np.zeros(log_lambda.size, dtype=bool_)
-        w2 = np.zeros(log_lambda.size, dtype=bool_)
-        return (log_lambda, flux, ivar, transmission_correction, mean_snr, bins,
-                rebin_ivar, orig_ivar, w1, w2)
 
     bins = find_bins(log_lambda, log_lambda_grid, wave_solution)
+    binned_arr_size = bins.max() + 1
     log_lambda = log_lambda_grid[0] + bins * pixel_step
```

Comment:
Yes, I think only out-of-boundary pixels are removed from spectra at this points. That is there are separate `w1` and `ivar>0` masks.

Reply:
Can you double check what you get when e.g. using a Delta_lambda=0.7 on DESI data with the normal spacing, will this produce an even grid with that spacing or have gaps in between. I'd guess the latter, but it's not super relevant right now (might be when adding other datasets at a later time?). If the former it would be interesting to know how the gaps are filled...